### ISSUE TITLE
Enable Smart Card Enforcement

### ISSUE THREAD

**Issue by ntwkninja** [defenseunicorns/uds-core#684]
### Is your feature request related to a problem? Please describe.
Currently, when a user authenticates with userpass+MFA or Smart Card Auth (CAC) there is nothing that distinguishes which auth method was used. Keycloak can service an environment that allows userpass+MFA and a separate environment that requires CAC / Smart Card login. 

We need a way to permit / deny traffic based on a user's authentication method (i.e. a mission app deployed to an environment that requires CAC auth but leverages the Keycloak IdP that serves a broader audience).

### Describe the solution you'd like

- **Given** a user authenticates to keycloak using userpass+MFA authentication where both authentication types are enabled
- **When** CAC auth is required for a particular environment
- **Then** access is denied

### Additional context
This could be a platform capability, the onus of a mission app (i.e. a JWT attribute is presented indicating auth method) or both.

**Comment by mjnagel** [comment]
We do (now) support [configurable authentication flows](https://uds.defenseunicorns.com/reference/uds-core/idam/authentication-flows/) which allow enabling any or all of user/pass, CAC, and SSO via another IdP.

This is a realm wide setting, so it would cover the entire environment (all clients logging into a given environment). I'm not sure this entirely helps in your scenario with a Keycloak IdP, but wanted to update this issue and mention it.

**Comment by slaskawi** [comment]
@mjnagel @ntwkninja Technically this can be done by duplicating the Authentication Flow, customizing it and pinning such a customized flow to specific clients (Client have a configuration attribute to use a specific Authentication Flow). 

Having in mind such an advanced configuration, I think this should be done via OpenTofu and not directly in UDS Core. Therefore, I propose closing this in favor of https://github.com/defenseunicorns/uds-identity-config/issues/530

**Comment by mjnagel** [comment]
@slaskawi is using a custom/specific auth flow something that could be done via the Package CR today? No concerns with having the "duplicated/custom auth flow config" be handled by the Tofu issue, just thinking about the "consumption" of that flow on the client side and how we'd expect that to be done.

**Comment by slaskawi** [comment]
@mjnagel This is definitely something to be done with OpenTofu. We probably don't want to provide this functionality out of the box as it's too error-prone.

**Comment by mjnagel** [comment]
> [@mjnagel](https://github.com/mjnagel) This is definitely something to be done with OpenTofu. We probably don't want to provide this functionality out of the box as it's too error-prone.

I think I largely agree with this, although I have one small hesitation:
> Client have a configuration attribute to use a specific Authentication Flow

Maybe this is my misunderstanding of the previous comment, but to properly set this up would a given client need modifications? Or is everything controlled on the auth flow side? _IF_ there is some configuration (attribute) on the client side, does the Package CR sso spec already support this? If not, what would we need to add to enable this functionality?

**Comment by ntwkninja** [comment]
> to properly set this up would a given client need modifications?

Yes, this is the needed client setting if using the tofu provider `authentication_flow_binding_overrides.browser_id`
